src/Core/Configurations/RuntimeConfigValidator.cs

```diff
@@ -209,9 +207,117 @@ public async Task<JsonSchemaValidationResult> ValidateConfigSchema(RuntimeConfig
+                    }
+
+                    // In one-one, one-many, or many-one relationships when both source and target are non null their size must match.
+                    if ((relationship.SourceFields is not null && relationship.TargetFields is not null) &&
+                        relationship.SourceFields.Length != relationship.TargetFields.Length)
+                    {
+                        HandleOrRecordException(new DataApiBuilderException(
+                            message: $"Entity: {entityName} has a relationship: {relationshipName}, which has {relationship.SourceFields.Length} source fields defined, " +
+                                $"but {relationship.TargetFields.Length} target fields defined.",
+                            statusCode: HttpStatusCode.ServiceUnavailable,
+                            subStatusCode: DataApiBuilderException.SubStatusCodes.ConfigValidationError));
+                    }
+                }
+
+                // Linking object exists and we therefore have a many-many relationship. Validation here differs from one-many and many-one in that
+                // the source and target fields are now only indirectly related through the linking object. Therefore, it is the source and linkingSource
+                // along with the target and linkingTarget fields that must match, respectively. Source and linkingSource fields provide the relationship
+                // from the source entity to the linkingObject while target and linkingTarget fields provide the relationship from the target entity to the
+                // linkingObject.
+                // see: https://learn.microsoft.com/en-us/azure/data-api-builder/relationships#many-to-many-relationship
+                if (!string.IsNullOrWhiteSpace(relationship.LinkingObject))
+                {
+                    ValidateFieldsAndAssociatedLinkingFields(
+                        fields: relationship.SourceFields,
+                        linkingFields: relationship.LinkingSourceFields,
+                        fieldType: "source",
+                        entityName: entityName,
+                        relationshipName: relationshipName);
+                    ValidateFieldsAndAssociatedLinkingFields(
+                        fields: relationship.TargetFields,
+                        linkingFields: relationship.LinkingTargetFields,
+                        fieldType: "target",
+                        entityName: entityName,
+                        relationshipName: relationshipName);
+                }
+            }
+        }
+    }
+
+    /// <summary>
+    /// This method validates the entities relationships against the database objects using
+    /// metadata from the backend DB generated by this function.
+    /// </summary>
+    public async Task ValidateEntityRelationshipsInConfigWithGeneratedMetadata(RuntimeConfig runtimeConfig, ILoggerFactory loggerFactory)
```

Comment:
```suggestion
    public async Task ValidateEntityRelationshipsInConfigWithDbMetadataAsync(RuntimeConfig runtimeConfig, ILoggerFactory loggerFactory)
```
name should be consistent with other validation "With/Without + DbMetadata" and also have "Async Suffix because it's an async method.

Reply:
Also, this method doesn't just validate entity relationships. there are validations in the metadataprovider.initialize that aren't related to relationships